server/src-lib/Hasura/RQL/Types/Common.hs

```diff
@@ -184,3 +189,16 @@ data ForeignKey
 $(deriveJSON (aesonDrop 3 snakeCase) ''ForeignKey)
 
 instance Hashable ForeignKey
+
+newtype GraphQLName
+  = GraphQLName {unGraphQLName :: G.Name}
+  deriving (Show, Eq, Lift, ToJSON)
+
+instance FromJSON GraphQLName where
```

Comment:
Replying from Slack:

> Hey, I don't think we should have a `FromJSON` parser to `G.Name` in `graphql-parser` library.  These are my thoughts; GraphQL is separate DSL which doesn't require to be parsable from JSON. Just like having a `FromJSON` instance to a SQL literal/identifier. It makes sense if we have graphql-parser within graphql-engine package. It's just my opinion 🙂

That’s a fair point, but I think I ultimately disagree, mainly because `G.Name` *already has* a `FromJSON` instance in `graphql-parser`, it just doesn’t check for validity of the name. I think that’s bad API design, since you want `FromJSON` instances to reject invalid input—that’s one of the main points of them—but the current one doesn’t.

I think that if `G.Name` didn’t have a `FromJSON` instance at all, your argument might make sense. So I think we ought to do one of two things:

  1. Replace the `FromJSON G.Name` instance in the `graphql-parser` package with this one.

  2. Remove the `FromJSON G.Name` instance in the `graphql-parser` package entirely.

I’m okay with either one, but I’d prefer the first option, just because I think this is a pretty canonical instance, anyway.

Reply:
Sorry, it's my mistake. I never knew there were `FromJSON` instances to GraphQL syntax in `graphql-parser-hs` library. I'll do `1`.